exporter/exporterparser/datadog.go

```diff
@@ -18,50 +18,46 @@ import (
 	"context"
 
 	datadog "github.com/DataDog/opencensus-go-exporter-datadog"
+	"github.com/spf13/viper"
 
 	"github.com/census-instrumentation/opencensus-service/data"
 	"github.com/census-instrumentation/opencensus-service/exporter"
 )
 
 type datadogConfig struct {
 	// Namespace specifies the namespaces to which metric keys are appended.
-	Namespace string `yaml:"namespace,omitempty"`
+	Namespace string `mapstructure:"namespace,omitempty"`
 
 	// TraceAddr specifies the host[:port] address of the Datadog Trace Agent.
 	// It defaults to localhost:8126.
-	TraceAddr string `yaml:"trace_addr,omitempty"`
+	TraceAddr string `mapstructure:"trace_addr,omitempty"`
 
 	// MetricsAddr specifies the host[:port] address for DogStatsD. It defaults
 	// to localhost:8125.
-	MetricsAddr string `yaml:"metrics_addr,omitempty"`
+	MetricsAddr string `mapstructure:"metrics_addr,omitempty"`
 
 	// Tags specifies a set of global tags to attach to each metric.
-	Tags []string `yaml:"tags,omitempty"`
+	Tags []string `mapstructure:"tags,omitempty"`
 
-	EnableTracing bool `yaml:"enable_tracing,omitempty"`
-	EnableMetrics bool `yaml:"enable_metrics,omitempty"`
+	EnableTracing bool `mapstructure:"enable_tracing,omitempty"`
+	EnableMetrics bool `mapstructure:"enable_metrics,omitempty"`
 }
 
 type datadogExporter struct {
 	exporter *datadog.Exporter
 }
 
-// DatadogTraceExportersFromYAML parses the yaml bytes and returns an exporter.TraceExporter targeting
+// DatadogTraceExportersFromViper unmarshals the viper and returns an exporter.TraceExporter targeting
 // Datadog according to the configuration settings.
-func DatadogTraceExportersFromYAML(config []byte) (tes []exporter.TraceExporter, mes []exporter.MetricsExporter, doneFns []func() error, err error) {
+func DatadogTraceExportersFromViper(v *viper.Viper) (tes []exporter.TraceExporter, mes []exporter.MetricsExporter, doneFns []func() error, err error) {
 	var cfg struct {
-		Exporters *struct {
```

Comment:
Does this radically change what the DataDog exporter will look like in the YAML file?

Reply:
No, it keeps it the same